### PR TITLE
Patterns

### PR DESCRIPTION
**Pull request by tvv923**
Patterns

### CODE REVIEW

File: jp/src/edu/hillel/lesson23/Factory/FurnitureFactory.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package edu.hillel.lesson23.Factory;
+
+public class FurnitureFactory {
```
Comment (kartius) [inline jp/src/edu/hillel/lesson23/Factory/FurnitureFactory.java]: It's a not an AbstractFactory. You need to make it as abstract class or interface and create a couple of factory imlementations

File: jp/src/edu/hillel/lesson23/Builder/CarBuilder.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package edu.hillel.lesson23.Builder;
+
+public class CarBuilder {
```
Comment (kartius) [inline jp/src/edu/hillel/lesson23/Builder/CarBuilder.java]: Usually, builder is creating inside a class as inner class.
There is a not mistake, just a best practices

File: jp/src/edu/hillel/lesson23/Factory/FurnitureFactory.java (hunk 1 above)
Comment (tvv923) [reply jp/src/edu/hillel/lesson23/Factory/FurnitureFactory.java]: Changed

File: jp/src/edu/hillel/lesson23/Builder/CarBuilder.java (hunk 2 above)
Comment (tvv923) [reply jp/src/edu/hillel/lesson23/Builder/CarBuilder.java]: Changed

**Review by kartius — APPROVED** [review]

**Pull request merged**